### ISSUE TITLE
vscode: Unexpected insertion of semicolons

### ISSUE THREAD

**Issue by NigelBreslaw** [slint-ui/slint#5966]
When writing code the intellisense to complete a property name also inserts a semicolon at the end of the line. This conflicts with how the default tooling (js/tsc) works in vscode.

There semicolons are added by the formatter only, not the intellisense. It's also inconsistent, so completions of properties e.g. 'green' don't add the semicolon. It makes editing feel unpredictable and you constantly end up with 2 semicolons at the end of the line when you auto just add them yourself. There is no way to build muscle memory if like me you are constantly swapping between slint files and javascript (a supported slint language) where the tooling works differently. 

tl;dr: Please don't intellisense insert semicolons. If a nice auto way to add them is needed leave it to the formatter that runs when the user want it to e.g. on save or via the menu.

**Comment by ogoffart** [comment]
Contrary to javascript, semicolons are mendatory in Slint.

The code that insert the semicolon is there:
https://github.com/slint-ui/slint/blob/039f33eaaeb2c5eeae4b52ebaecedbb410fc2e73/tools/lsp/language/completion.rs#L72

What this code does is that when you type, say (where `|` is the cursor)

```slint
back|
```

and autocomplete background, it will insert

```slint
background: |;
```

While putting the cursor before the semicolon.
(so it doesn't do it when you type `green`, but the semicolon is added before)

I have been personally bitten by the double semicolon before and I'd not be opposed to remove it. What do other think ?

**Comment by NigelBreslaw** [comment]
Can the functionality move to the formatter? In many places that would also auto fix a bunch of missing semi colons you end up adding. You as in me, a javascripter :-D

**Comment by ogoffart** [comment]
> Can the functionality move to the formatter?

No. As I said, semicolons are mandatory. This is not a formatting issue, this is an error to not have the semicolon.
